src/sas/qtgui/MainWindow/DataExplorer.py

```diff
@@ -1090,13 +1091,14 @@ def displayData(self, data_list, id=None):
 
             plot_name = plot_to_show.name
             role = plot_to_show.plot_role
+            stand_alone_types = [DataRole.ROLE_RESIDUAL, DataRole.ROLE_STAND_ALONE]
 
-            if (role == Data1D.ROLE_RESIDUAL and shown) or role == Data1D.ROLE_DELETABLE:
+            if (role in stand_alone_types and shown) or role == DataRole.ROLE_DELETABLE:
                 # Nothing to do if separate plot already shown or to be deleted
                 continue
-            elif role == Data1D.ROLE_RESIDUAL:
-                # Residual plots should always be separate
-                plot_to_show.yscale='linear'
+            elif role in stand_alone_types:
+                # Stand-alone linear and residual plots should always be separate
+                plot_to_show.yscale = 'linear'
```

Comment:
I believer line 1101 is either overwriting or being overwritten (that is part of the problem, even If I know there are two ways to achieve something I never know which one wins) by `FittingUtilities` function `residualsData1D` where line 550 has
```python
 residuals.ytransform = 'y'
```
actually in building that Data1D object, the axes labels are defined and the x axis is made linear for sesans data using the same transform on lines 551 and beyond
```python
    if reference_data.isSesans:
        residuals.xtransform = 'x'
        residuals.xaxis('\\rm{z} ', 'A')
    # For latter scale changes
    else:
        residuals.xaxis('\\rm{Q} ', 'A^{-1}')
    residuals.yaxis('\\rm{Residuals} ', 'normalized')
```
So it seems that the plottables Data1D object (I guess that was the idea of a plottable?) has all the hooks for defining how that data should be plotted while the plotting itself has functions to do the same thing? I'm not sure what the right answer should be but maybe that is monstrous can of worm?

If so then the easiest is to stick to using the Data1D hooks for everything in which case we just delete this line?

Also, shouldn't the comment be something like "identify plots that should be separate"?

Reply:
I tested this, and residual, polydispersity, and P(r) plots all continue to plot with a linear y scale after commenting out line 1101, so I removed it.